src/burnchains/events.rs

```diff
@@ -123,6 +168,14 @@ where
     }
 }
 
+// Only works if Display implementation uses a hex string, which Txid and StacksBlockId do
+fn ser_as_hexstr<T: Display, S: Serializer>(input: &T, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    serializer.serialize_str(&format!("0x{}", &input.to_string()))
```

Comment:
You can avoid a `String` instantiation by just using the display fmt

```suggestion
    serializer.serialize_str(&format!("0x{}", &input))
```